src/main/clojure/cemerick/pomegranate/aether.clj

```diff
@@ -637,28 +626,68 @@ kwarg to the repository kwarg.
      (= (:classifier om)
         (:classifier som)))))
 
-(defn- find-managed-coord
-  "Given an entry from a coordinates vector, and a managed coordinates vector, find
-  the entry in the managed coordinates vector (if any) that matches the coordinate."
-  [coord managed-coords]
-  (first (filter #(coordinates-match? coord %) managed-coords)))
-
-(defn- add-version-from-managed-coords-if-missing
-  "Given a managed coordinates vector and an entry from a coordinates vector, check
-  to see if the coordinate specifies a version string, and if not, update it with
-  the version string from the managed coordinates (if it is defined)."
-  [managed-coords coord]
-  (if (nil? (second coord))
-    (add-version-from-managed-coord coord (find-managed-coord coord managed-coords))
-    coord))
+(defn- coord-map-key
+  "Creates the unique key for the given coordinate map.  This is
+   essentially a representation of the full maven coordinate except
+   the version."
+  [{:keys [project extension classifier]}]
+  [project extension classifier])
+
+(defn- canonical-id [id]
+  (apply symbol (distinct ((juxt group name) id))))
+
+(defn- conform-coord
+  "Returns a map describing the coordinate.  The full project ID is
+   under the :project key and the version, if specified and not nil,
+   under the :version key.  All other specified options appear with
+   the keys and values specified (including nil values)."
+  [[project & opts]]
+  (if project
+    (let [has-version? (odd? (count opts))
```

Comment:
A more accurate way to do this would be to fully parse the dependency vector.  Doing this by hand is a lot of work for little gain.  I think the above works for all valid dependency vectors with/without versions, but if you can think of cases where it doesn't I'm happy to change this. 

As an aside, I have (alpha) spec definitions for the dependency vectors.  I didn't think of adding them here as they would require Clojure 1.9+.  However, I can put them in if they are useful, although I'd probably need help in creating the conditional build/test with Clojure 1.9.

```clojure
(ns deps.spec
  (:require [clojure.spec :as s]))

(s/def ::extension string?)
(s/def ::classifier string?)
(s/def ::scope #{"compile" "test" "provided"})
(s/def ::optional boolean?)

(s/def ::exclusion
  (s/& (s/cat :project symbol?
              :options (s/keys* :opt-un [::extension ::classifier]))
       (s/conformer (fn [{:keys [project options]}] (merge options {:project project}))
                    (fn [{:keys [project] :as m}] (let [m (dissoc m :project)]
                                                    {:project project
                                                     :options m})))))

(s/def ::exclusions (s/coll-of ::exclusion :min-count 1))

(s/def ::dep
  (s/& (s/cat :project symbol?
              :version string?
              :options (s/keys* :opt-un [::extension ::classifier ::scope ::optional ::exclusions]))
       (s/conformer (fn [{:keys [project version options]}]
                      (merge options {:project project, :version version}))
                    (fn [{:keys [project version] :as m}]
                      (let [m (dissoc m :project :version)]
                        {:project project
                         :version version
                         :options m})))))
```